webapp/src/components/post_type/action_view/action_view.jsx

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import ActionButton from './action_button';
+
+export default class ActionView extends React.PureComponent {
+    static propTypes = {
+        post: PropTypes.object.isRequired,
+        attachment: PropTypes.object.isRequired,
+        votedAnswers: PropTypes.object.isRequired,
+        siteUrl: PropTypes.string.isRequired,
+
+        actions: PropTypes.shape({
+            fetchVotedAnswers: PropTypes.func.isRequired,
+        }).isRequired,
+    }
+
+    componentDidMount() {
+        this.props.actions.fetchVotedAnswers(this.props.siteUrl, this.props.post.props.poll_id);
+    }
+
+    render() {
+        const actions = this.props.attachment.actions;
+        if (!actions || !actions.length) {
+            return '';
+        }
+
+        const content = [];
+        const votedAnswers = this.props.votedAnswers || {};
+        const answers = votedAnswers[this.props.post.props.poll_id] || {};
+
+        actions.forEach((action) => {
+            if (!action.id || !action.name) {
```

Comment:
Could possibly filter instead of a forEach with if statement.
e.g. `actions.filter( (action) => action.id && action.name ).forEach( (action) ...`

Reply:
It looks smart 👍